apollo-router/src/plugins/demand_control/mod.rs

```diff
@@ -163,6 +166,10 @@ impl IntoGraphQLErrors for DemandControlError {
                 .extension_code(self.code())
                 .message(self.to_string())
                 .build()]),
+            DemandControlError::FederationError(_) => Ok(vec![graphql::Error::builder()
+                .extension_code(self.code())
+                .message(self.to_string())
+                .build()]),
```

Comment:
The only two errors coming from federation would be undefined directive definitions, right? do we want to consider that a federation error or a graphql validation error?

Reply:
The only case this should arise is when we get an unexpected failure in `spec.directive_name_in_schema` during execution. Realistically, this should never actually arise as an execution error because we call the same thing during subgraph extraction on startup. I don't think the spec definitions guarantees that error to specifically be a validation issue, so we probably need to keep the name here generic.